packages/appstore/src/components/modals/wallet-modal/wallet-modal.tsx

```diff
@@ -18,13 +18,13 @@ const WalletModal = observer(() => {
 
     useEffect(() => {
         let timeout_id: NodeJS.Timeout;
-        if (wallet?.loginid !== active_modal_wallet_id) {
+        if (is_wallet_modal_visible && wallet?.loginid !== active_modal_wallet_id) {
             /** Adding a delay as per requirement because the modal must appear first, then switch the account */
             timeout_id = setTimeout(() => switchAccount(active_modal_wallet_id), 700);
         }
 
         return () => clearTimeout(timeout_id);
-    }, [active_modal_wallet_id, switchAccount, wallet?.loginid]);
+    }, [active_modal_wallet_id, is_wallet_modal_visible, switchAccount, wallet?.loginid]);
```

Comment:
Does it still work if you add the `is_wallet_modal_visible` before `switchAccount` like this:

```
    useEffect(() => {
        let timeout_id: NodeJS.Timeout;
        if (wallet?.loginid !== active_modal_wallet_id) {
            /** Adding a delay as per requirement because the modal must appear first, then switch the account */
            timeout_id = setTimeout(() => is_wallet_modal_visible && switchAccount(active_modal_wallet_id), 700);
        }

        return () => clearTimeout(timeout_id);
    }, [active_modal_wallet_id, switchAccount, wallet?.loginid]);
```

Reply:
@aizad-deriv I think yes, it will work, just will check `is_wallet_modal_visible` after 700 milliseconds instead of checking at the component render. But honestly I don't understand why we need to refactor like this?